examples/closure-app/silabs/src/ClosureManager.cpp

```diff
@@ -230,34 +305,228 @@ ClosureManager::OnMoveToCommand(const chip::Optional<chip::app::Clusters::Closur
-        isCalibrationInProgress = false;
-        break;
+        ChipLogError(AppServer, "SetTarget action is already in progress on Endpoint %d", endpointId);
+        return Status::Failure;
     }
-    case Action_t::STOP_MOTION_ACTION:
-        // This should handle the completion of a stop motion action.
-        break;
-    case Action_t::STOP_CALIBRATE_ACTION:
-        // This should handle the completion of a stop calibration action.
-        break;
-    case Action_t::MOVE_TO_ACTION:
-        // This should handle the completion of a move-to action.
-        break;
-    default:
-        ChipLogError(AppServer, "Invalid action received in HandleClosureAction");
-        break;
+
+    // Update OverallTarget of Closure based on SetTarget command.
+    DataModel::Nullable<GenericOverallTargetState> overallTargetState;
+    VerifyOrReturnError(ep1.GetLogic().GetOverallTargetState(overallTargetState) == CHIP_NO_ERROR, Status::Failure,
+                        ChipLogError(AppServer, "Failed to get overall target for SetTarget command"));
+
+    if (overallTargetState.IsNull())
+    {
+        overallTargetState.SetNonNull(GenericOverallTargetState{});
     }
-    // Reset the current action after handling the closure action
-    GetInstance().SetCurrentAction(Action_t::INVALID_ACTION);
+
+    if (position.HasValue())
+    {
+        // Set overallTargetState position to NullOptional as panel position change cannot be represented in OverallTarget.
+        overallTargetState.Value().position.SetValue(DataModel::NullNullable);
+    }
+
+    if (latch.HasValue())
+    {
+        overallTargetState.Value().latch.SetValue(DataModel::MakeNullable(latch.Value()));
+    }
+
+    if (speed.HasValue())
+    {
+        overallTargetState.Value().speed.SetValue(speed.Value());
+    }
+
+    VerifyOrReturnError(ep1.GetLogic().SetOverallTargetState(overallTargetState) == CHIP_NO_ERROR, Status::Failure,
+                        ChipLogError(AppServer, "Failed to set overall target for SetTarget command for Endpoint %d", endpointId));
+
+    VerifyOrReturnError(ep1.GetLogic().SetCountdownTimeFromDelegate(kCountdownTimeSeconds) == CHIP_NO_ERROR, Status::Failure,
+                        ChipLogError(AppServer, "Failed to set countdown time for move to command on Endpoint %d", endpointId));
+
+    VerifyOrReturnError(ep1.GetLogic().SetMainState(MainStateEnum::kMoving) == CHIP_NO_ERROR, Status::Failure,
+                        ChipLogError(AppServer, "Failed to set main state for move to command on Endpoint 1"));
+
+    // Post an event to initiate the unlatch action asynchronously.
+    // Closure panel first performs the unlatch action if it is currently latched,
+    // and then continues with the SetTarget action.
+    // This is to ensure that the panel can move to the target position without being latched.
+    SetCurrentAction(Action_t::PANEL_UNLATCH_ACTION);
+    mCurrentActionEndpointId = endpointId;
+    isSetTargetInProgress    = true;
```

Comment:
should there be some thread safetyness mechanism around those members of the instance